source/tests/utilities/client_helpers.h

```diff
@@ -55,10 +55,34 @@ class simple_variant {
   int8_t slot_{-1};
 };
 
-inline void raise_if_error(const ::grpc::Status& status)
+class grpc_driver_error : public std::runtime_error {
+ private:
+  ::grpc::StatusCode code_;
+  std::multimap<std::string, std::string> trailers_;
+
+ public:
+  grpc_driver_error(const std::string& message, ::grpc::StatusCode code, const std::multimap<grpc::string_ref, grpc::string_ref>& trailers)
+      : std::runtime_error(message), code_(code)
+  {
+    for (auto trailer : trailers) {
+      trailers_.emplace(trailer.first.data(), trailer.second.data());
+    }
+  }
+  ::grpc::StatusCode StatusCode() const
```

Comment:
FYI: C# [RpcException](https://github.com/grpc/grpc-dotnet/blob/master/src/Grpc.Core.Api/RpcException.cs) also includes the `Status` object, but I guess this object has enough info to reconstruct it.

Reply:
Yeah, and I'm trying to limit unnecessary copy operations. It's something we could always add later if needed.